### PR TITLE
adding psalm

### PR DESCRIPTION
**Pull request by ArtificialOwl**
also some php 8 syntax

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]

File: lib/Lookup.php [hunk 1]
```diff
@@ -19,52 +22,24 @@
...
+		private IClientService $clientService,
+		private LoggerInterface $logger,
+		private ICloudIdManager $cloudIdManager,
+		IConfig $config
 	) {
-		$this->httpClientService = $clientService;
-		$this->lookupServerUrl = $config->getSystemValue('lookup_server', '');
-		$this->logger = $logger;
-		$this->cloudIdManager = $cloudIdManager;
+		$this->lookupServerUrl = $config->getSystemValue('lookup_server', '') ?? '';
```
Comment (come-nc) [inline lib/Lookup.php:42]:
```suggestion
		$this->lookupServerUrl = $config->getSystemValueString('lookup_server', '');
```

File: lib/UserDiscoveryModules/ManualUserMapping.php [hunk 2]
```diff
@@ -42,25 +42,17 @@
...
+	private string $file;
+	private bool $useRegularExpressions;
+
+	public function __construct(
+		IConfig $config,
+		private LoggerInterface $logger
+	) {
 		$this->idpParameter = $config->getSystemValue('gss.discovery.manual.mapping.parameter', '');
 		$this->file = $config->getSystemValue('gss.discovery.manual.mapping.file', '');
 		$this->useRegularExpressions = $config->getSystemValue('gss.discovery.manual.mapping.regex', false);
```
Comment (come-nc) [inline lib/UserDiscoveryModules/ManualUserMapping.php:55]:
```suggestion
		$this->idpParameter = $config->getSystemValueString('gss.discovery.manual.mapping.parameter', '');
		$this->file = $config->getSystemValueString('gss.discovery.manual.mapping.file', '');
		$this->useRegularExpressions = $config->getSystemValueBool('gss.discovery.manual.mapping.regex', false);
```

File: tests/stub.phpstub [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    class SchemaWrapper {
+        public function __construct(IDBConnection $connection) {}
+
+        public function performDropTableCalls();
+    }
+}
+
+namespace OC {
+    class SystemConfig {}
+}
```
Comment (come-nc) [inline tests/stub.phpstub:107]: You cannot use OCP\IConfig?

File: tests/stub.phpstub [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+namespace Doctrine\DBAL\Query {
+    class QueryBuilder {
+        public const SELECT = 'select';
+    }
+}
+
+namespace OC\DB\QueryBuilder {
```
Comment (come-nc) [inline tests/stub.phpstub:115]: Why do you need this one? (OCP\IQueryBuilder should be enough)

**Review by come-nc — COMMENTED** [review]
Also note that `\OCP\HintException` is available if you need hinted exceptions.

File: tests/stub.phpstub (hunk 4 above)
Comment (ArtificialOwl) [reply tests/stub.phpstub]: fixed

**Pull request merged**